### PR TITLE
[HLS-2906] Sarama proxy library

### PR DESCRIPTION
**Pull request by lior-govrin**

### CODE REVIEW

File: proxy-libs/heliossarama/sarama.go [hunk 1]
```diff
@@ -0,0 +1,875 @@
...
+
+func WithCustomFallbackPartitioner(randomHP Partitioner) HashPartitionerOption {
+	return originalSarama.WithCustomFallbackPartitioner(randomHP)
+}
+
+func WithCustomHashFunction(hasher func() hash.Hash32) HashPartitionerOption {
+	return originalSarama.WithCustomHashFunction(hasher)
+}
+
+// TODO.
```
Comment (nozik) [inline proxy-libs/heliossarama/sarama.go:872]: ??

File: proxy-libs/heliossarama/sarama_test.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	asyncProducer.Close()
+
+	consumerGroup, _ := NewConsumerGroup(Addresses, "consumerGroup", config)
+	consumerGroup.Consume(context.Background(), []string{Topic}, &TestConsumerGroupHandler{
+		messageKey:   key,
+		messageValue: value,
+		spanRecorder: spanRecorder,
+		t:            t,
+	})
+	// consumerGroup.Close()
```
Comment (nozik) [inline proxy-libs/heliossarama/sarama_test.go:129]: no need to close?

File: proxy-libs/heliossarama/sarama_test.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+	client, _ := NewClient(Addresses, config)
+	consumerGroup, _ := NewConsumerGroupFromClient("consumerGroupFromClient", client)
+	consumerGroup.Consume(context.Background(), []string{Topic}, &TestConsumerGroupHandler{
+		messageKey:   key,
+		messageValue: value,
+		spanRecorder: spanRecorder,
+		t:            t,
+	})
+	// consumerGroup.Close()
```
Comment (nozik) [inline proxy-libs/heliossarama/sarama_test.go:155]: no need to colse?

File: proxy-libs/heliossarama/sarama_test.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+func (handler *TestConsumerGroupHandler) Setup(session ConsumerGroupSession) error {
+	return nil
+}
+
+func (handler *TestConsumerGroupHandler) Cleanup(session ConsumerGroupSession) error {
+	return nil
+}
+
+func (handler *TestConsumerGroupHandler) ConsumeClaim(session ConsumerGroupSession, claim ConsumerGroupClaim) error {
+	message := <-claim.Messages()
```
Comment (nozik) [inline proxy-libs/heliossarama/sarama_test.go:43]: Can you create a span here and check that its parent is the consume span?

**Review by nozik — CHANGES_REQUESTED** [review]

File: proxy-libs/heliossarama/sarama_test.go (hunk 4 above)
Comment (lior-govrin) [reply proxy-libs/heliossarama/sarama_test.go:43]: Same problem with the context propagation here - I have to manually extract the context from the message (with `otelsarama`), and pass it on, in the test itself - So I wouldn't be checking anything really.

**Review by nozik — APPROVED** [review]

**Pull request merged**